pages/components/Collection/Collection.js

```diff
@@ -1,5 +1,94 @@
-const Collection = () => (
-  <h1>123</h1>
-)
+import styles from "@/pages/components/Collection/Collection.module.scss"
+import React, { useState } from "react"
+import Image from "next/image"
+import wineImage from "../../../public/wineImage.png"
+import circlesImage from "../../../public/circle.svg"
+import circles_active from "../../../public/circle_active.svg"
+import prevItemImage from "../../../public/prevItem.png"
+import nextItemImage from "../../../public/nextItem.png"
+import { wines, wineType } from "@/pages/utils/varibles"
+
+const Collection = () => {
+  const [ winesCollection, setWinesCollection ] = useState(wines)
+  const [ currentIndex, setCurrentIndex ] = useState(0)
+  const [ activeButton, setActiveButton ] = useState(null)
+
+  const sort = (e) => {
+    setCurrentIndex(0)
+    setWinesCollection(wines.filter((element) => {
+      return element.classification === e
+    }))
+  }
+
+  const changeColor = (index) => {
+    setActiveButton(index)
+  }
+
+  const nextItem = () => {
+    setCurrentIndex((prevIndex) => (prevIndex + 1) % winesCollection.length)
+  }
+
+  const prevItem = () => {
+    setCurrentIndex((prevIndex) => (prevIndex - 1 + winesCollection.length) % winesCollection.length)
+  }
+
+  const circles = Array.from({ length: winesCollection.length < 5 ? winesCollection.length : 5 }, (_, index) => (
```

Comment:
Calculate values separately from the place where it is used. It makes reading the code simpler.

`const arrayLength = winesCollection.length < 5 ? winesCollection.length : 5`

`Array.from({ length: arrayLength})`

Reply:
Thanks, fixed